esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -713,47 +713,43 @@ bool CFileSprayEx::onDFUWUSearch(IEspContext &context, IEspDFUWUSearchRequest &
     return true;
 }
 
-int readFromCommaSeparatedString(const char *commaSeparatedString, StringBuffer* output)
+//Sasha List command returns comma separated strings about archived DFU WUs.
+//Each resultStr is for one DFU WU: WUID,User,JobName,ClusterName,StateMessage,Command.
+//Parse the resultStr to build one DFU WU and append it to the DFU WU list.
+void CFileSprayEx::appendDFUWUFromSashaListResult(const char *resultStr, IArrayOf<IEspDFUWorkunit> &dfuWUs)
 {
-    int numOfItems = 0;
-    if (commaSeparatedString && *commaSeparatedString)
-    {
-        char *pStr = (char *) commaSeparatedString;
-        while (pStr)
-        {
-            char item[1024];
-            bool bFoundComma = false;
-            int len = strlen(pStr);
-            for (int i = 0; i < len; i++)
-            {
-                char *pStr1 = pStr + i;
-                if (pStr1[0] != ',')
-                    continue;
+    StringArray wuInfoFields;
+    wuInfoFields.appendList(resultStr, ",");
+    unsigned numOfWUInfoFields = wuInfoFields.length();
+    if (numOfWUInfoFields == 0)
+        return;
 
-                strncpy(item, pStr, pStr1 - pStr);
-                item[pStr1 - pStr] = 0;
+    Owned<IEspDFUWorkunit> dfuWU = createDFUWorkunit();
+    dfuWU->setArchived(true);
+    dfuWU->setID(wuInfoFields.item(0));
+    if (numOfWUInfoFields == 1)
```

Comment:
If any fields are missing, you return without adding the IEspDFUWorkunit.
If that's the correct behavior then you can just return if numOfWUInfoFields is less than 6 at the very begining.

If you mean to support some fields missing, you shouldn't return rather you should skip all following tests.

If you want to allow some missing fields, you may want to have another function fill in the values, then if success this function can add the WU to the array.

Reply:
The question boils down to whether sasha ever leaves out any commas.  a,b,c,,,
Would be fine.  But if sasha leaves off a comma for a valid string a,b,c
then you need to deal with it and still create the WU?

Reply:
Sorry for my mistake. Sasha does leave out the commas. But, better keep the existing functionality.

Reply:
This is fine, and we could merge, but you could make it easier to read by doing a loop, something like:

```
  ForEachItemIn(i, wuInfoFields)
  {
     switch(i)
     {
         case 0:
            dfuWU->setID(wuInfoFields.item(i));
            break;
         case 1:
            dfuWU->setUser(wuInfoFields.item(i));
            break;
            ...
      }
      return dfuWU.getClear();
  }
```